nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/storage/ListAzureDataLakeStorage.java

```diff
@@ -123,33 +122,34 @@ public class ListAzureDataLakeStorage extends AbstractListProcessor<ADLSFileInfo
     public static final PropertyDescriptor PATH_FILTER = new PropertyDescriptor.Builder()
             .name("path-filter")
             .displayName("Path Filter")
-            .description(String.format("When '%s' is true, then only subdirectories whose paths match the given regular expression will be scanned", RECURSE_SUBDIRECTORIES.getDisplayName()))
+            .description("Only subdirectories whose paths match the given regular expression will be scanned")
             .required(false)
             .addValidator(StandardValidators.REGULAR_EXPRESSION_WITH_EL_VALIDATOR)
             .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+            .dependsOn(RECURSE_SUBDIRECTORIES, "true")
             .build();
 
     private static final List<PropertyDescriptor> PROPERTIES = Collections.unmodifiableList(Arrays.asList(
-            ADLS_CREDENTIALS_SERVICE,
-            FILESYSTEM,
-            DIRECTORY,
-            RECURSE_SUBDIRECTORIES,
-            FILE_FILTER,
-            PATH_FILTER,
-            RECORD_WRITER,
-            LISTING_STRATEGY,
-            TRACKING_STATE_CACHE,
-            TRACKING_TIME_WINDOW,
-            INITIAL_LISTING_TARGET));
+        FILESYSTEM,
+        DIRECTORY,
+        ADLS_CREDENTIALS_SERVICE,
+        FILE_FILTER,
+        RECURSE_SUBDIRECTORIES,
+        PATH_FILTER,
```

Comment:
I believe `DIRECTORY` and `RECURSE_SUBDIRECTORIES` belong together and should not be split. Also `FILE_FILTER` and `PATH_FILTER`.